src/esp/assets/PTexMeshData.cpp

```diff
@@ -572,67 +760,120 @@ void PTexMeshData::uploadBuffersToGPU(bool forceReload) {
   }
 
   for (int iMesh = 0; iMesh < submeshes_.size(); ++iMesh) {
-    LOG(INFO) << "\rLoading mesh " << iMesh + 1 << "/" << submeshes_.size()
-              << "... ";
+    LOG(INFO) << "Loading mesh " << iMesh + 1 << "/" << submeshes_.size()
+              << ". ";
 
     renderingBuffers_.emplace_back(
         std::make_unique<PTexMeshData::RenderingBuffer>());
 
     auto& currentMesh = renderingBuffers_.back();
-    currentMesh->vbo.setData(submeshes_[iMesh].vbo,
-                             Magnum::GL::BufferUsage::StaticDraw);
-    currentMesh->ibo.setData(submeshes_[iMesh].ibo,
-                             Magnum::GL::BufferUsage::StaticDraw);
+    currentMesh->vertexBuffer.setData(submeshes_[iMesh].vbo,
+                                      Magnum::GL::BufferUsage::StaticDraw);
+    currentMesh->indexBuffer.setData(submeshes_[iMesh].ibo,
+                                     Magnum::GL::BufferUsage::StaticDraw);
   }
-  LOG(INFO) << "... done" << std::endl;
 
   LOG(INFO) << "Calculating mesh adjacency... ";
-
   std::vector<std::vector<uint32_t>> adjFaces(submeshes_.size());
 
+  // load it if it is computed before.
+  // otherwise compute it once and save it for future usage.
+  const std::string adjFaceFilename =
+      Corrade::Utility::Directory::join(atlasFolder_, "../adjFaces.bin");
+  if (!loadAdjacency(adjFaceFilename, adjFaces)) {
 #pragma omp parallel for
-  for (int iMesh = 0; iMesh < submeshes_.size(); ++iMesh) {
-    calculateAdjacency(submeshes_[iMesh], adjFaces[iMesh]);
+    for (int iMesh = 0; iMesh < submeshes_.size(); ++iMesh) {
+      calculateAdjacency(submeshes_[iMesh], adjFaces[iMesh]);
+    }
+    // Warning: you should have enough disk space to store the info
+    // it usually takes a couple of 100MB (usually 200+MB).
+    saveAdjacency(adjFaceFilename, adjFaces);
+    LOG(INFO) << "Done: it is computed and saved to: " << adjFaceFilename;
+  } else {
+    LOG(INFO) << "Done: Loaded it from " << adjFaceFilename;
   }
 
+  // exit(-1);
+
+  int maxBufferTextureTexels = 0;
+  glGetIntegerv(GL_MAX_TEXTURE_BUFFER_SIZE, &maxBufferTextureTexels);
```

Comment:
There's `GL::BufferTexture::maxSize()` btw. ;) Also if you search for `GL_MAX_TEXTURE_BUFFER_SIZE` in magnum docs, it'll lead you to that function.